gateway/api/ray.py

```diff
@@ -210,11 +210,100 @@ def submit_job(job: Job) -> Job:
+    job_config.workers = job_config.workers or settings.RAY_CLUSTER_WORKER_REPLICAS
+    job_config.min_workers = (
+        job_config.min_workers or settings.RAY_CLUSTER_WORKER_MIN_REPLICAS
+    )
+    job_config.max_workers = (
+        job_config.max_workers or settings.RAY_CLUSTER_WORKER_MAX_REPLICAS
+    )
+    job_config.auto_scaling = (
+        job_config.auto_scaling or settings.RAY_CLUSTER_WORKER_AUTO_SCALING
+    )
+
+    # cpu job settings
+    node_selector_label = settings.RAY_CLUSTER_CPU_NODE_SELECTOR_LABEL
+
+    # if gpu job, use gpu nodes and resources
+    gpu_request = 0
+    if job.gpu:
+        node_selector_label = settings.RAY_CLUSTER_GPU_NODE_SELECTOR_LABEL
+        gpu_request = settings.LIMITS_GPU_PER_TASK
+
+    # configure provider configuration if needed
+    node_image = settings.RAY_NODE_IMAGE
+    provider_name = None
+    if job.program.provider is not None:
+        node_image = job.program.image
+        provider_name = job.program.provider.name
+
+    user_file_storage = FileStorage(
+        username=user.username,
+        working_dir=WorkingDir.USER_STORAGE,
+        function_title=job.program.title,
+        provider_name=provider_name,
+    )
+    provider_file_storage = user_file_storage
+    if job.program.provider is not None:
+        provider_file_storage = FileStorage(
+            username=user.username,
+            working_dir=WorkingDir.PROVIDER_STORAGE,
+            function_title=job.program.title,
+            provider_name=provider_name,
+        )
+
+    cluster = get_template("rayclustertemplate.yaml")
+    manifest = cluster.render(
+        {
+            "cluster_name": cluster_name,
+            "user_data_folder": user_file_storage.sub_path,
+            "provider_data_folder": provider_file_storage.sub_path,
+            "node_image": node_image,
+            "workers": job_config.workers,
+            "min_workers": job_config.min_workers,
+            "max_workers": job_config.max_workers,
+            "auto_scaling": job_config.auto_scaling,
+            "user": user.username,
+            "node_selector_label": node_selector_label,
+            "gpu_request": gpu_request,
+        }
+    )
+    cluster_data = yaml.safe_load(manifest)
+    return cluster_data
+
+
+def _is_local_mode():
+    local = settings.RAY_CLUSTER_MODE.get("local")
+    ray_local_host = settings.RAY_CLUSTER_MODE.get("ray_local_host")
+    return local and ray_local_host
+
+
+def _create_local_compute_resource(job: Job) -> ComputeResource:
+    compute_resource = ComputeResource.objects.filter(
+        host=settings.RAY_CLUSTER_MODE.get("ray_local_host")
+    ).first()
+    if compute_resource is None:
+        compute_resource = ComputeResource(
+            host=settings.RAY_CLUSTER_MODE.get("ray_local_host"),
+            title="Local compute resource",
+            owner=job.author,
+        )
+        compute_resource.save()
+    job.compute_resource = compute_resource
+    job.save()
```

Comment:
Reviewing the flow we agree that we can remove these two lines as they are being managed outside the function in the `schedule` and `schedule_queued_jobs`.